IOTConnectSDK/iotconnect-sdk_2_1/src/main/java/com/iotconnectsdk/mqtt/IotSDKMQTTService.kt

```diff
@@ -129,12 +151,84 @@ internal class IotSDKMQTTService private constructor(
             }
         })
         val mqttConnectOptions = MqttConnectOptions()
-        mqttConnectOptions.isAutomaticReconnect = false
+        mqttConnectOptions.isAutomaticReconnect = true
         mqttConnectOptions.isCleanSession = true
-        mqttConnectOptions.userName = protocolBean.un
-        if (protocolBean.pwd != null) {
-            mqttConnectOptions.password = protocolBean.pwd.toCharArray()
+
+        var sdkObj: JSONObject
+        try {
+            if (sdkOptions != null) {
+                sdkObj = JSONObject(sdkOptions)
+                if (authenticationType == IotSDKUrls.AUTH_TYPE_SELF_SIGN || authenticationType == IotSDKUrls.AUTH_TYPE_CA_SIGN) {
+                    if (sdkObj.has("certificate")) {
+                        val certificate = sdkObj.getJSONObject("certificate")
```

Comment:
@sw-tt-rajkiranvank @sw-tt-tusharojha  what happens if certificate name is different

Reply:
Hi @sw-ot-deveshmevada 
I have made sdk options and environment variable accessible from sdk only. Now he is not able to change keys